lametro/views.py

```diff
@@ -103,6 +106,29 @@ def extra_context(self):
 
         return extra
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        if "invalid_form" in kwargs:
+            context["alert_form"] = kwargs.get("invalid_form")
+        else:
+            context["alert_form"] = AlertForm()
+
+        return context
+
+    def post(self, request, *args, **kwargs):
```

Comment:
If you move validation to the form model, then you can extend [`form_valid()`](https://ccbv.co.uk/projects/Django/4.2/django.views.generic.edit/FormView/#form_valid) here, as you do in the AlertDelete view.

Reply:
Cool, got this to work as well! I could have overthought this, but without a post method, I was getting an error `Method Not Allowed (POST): /`. I found that to get past this and also get the `form_valid()` to run, I could have the LAMetroIndexView also inherit from the FormView. So that's what's in there now.

Is that kosher or should I change this?

Reply:
Kosher!